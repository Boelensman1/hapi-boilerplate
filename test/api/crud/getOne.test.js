const path = require('path')
const setUpCrudTest = require('test/api/setUpCrudTest')
const { role } = require('models')

const seedLocation = path.join(__dirname, 'getOne.seed.yaml')

const { getOne } = require('api/crud')

describe('Test getOne crud function', () => {
  test('Get one', async () => {
    const { request, h } = await setUpCrudTest(seedLocation)
    const handler = getOne('role', role).handler.bind(null, request, h)

    request.params = { id: 10 }

    const {
      body: { result },
    } = await handler()

    expect(result).toMatchSnapshot({
      createdAt: expect.any(Date),
    })
  })
  test('Get one with relations', async () => {
    const { request, h } = await setUpCrudTest(seedLocation)
    const handler = getOne('role', role).handler.bind(null, request, h)

    request.query = { getRelations: true }
    request.params = { id: 10 }

    const {
      body: { result },
    } = await handler()

    expect(result).toMatchSnapshot({
      createdAt: expect.any(Date),
      users: [
        {
          createdAt: expect.any(Date),
        },
      ],
    })
  })
})
