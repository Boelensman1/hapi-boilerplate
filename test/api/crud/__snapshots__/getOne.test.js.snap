// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test getOne crud function Get one 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": 10,
  "name": "nameC",
  "post_create": 0,
  "post_delete": 0,
  "post_read": 0,
  "post_update": 0,
  "role_create": 0,
  "role_delete": 0,
  "role_read": 0,
  "role_update": 0,
  "updatedAt": null,
  "user_create": 0,
  "user_delete": 0,
  "user_read": 0,
  "user_update": 0,
}
`;

exports[`Test getOne crud function Get one with relations 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": 10,
  "name": "nameC",
  "post_create": 0,
  "post_delete": 0,
  "post_read": 0,
  "post_update": 0,
  "role_create": 0,
  "role_delete": 0,
  "role_read": 0,
  "role_update": 0,
  "updatedAt": null,
  "user": Array [
    Object {
      "createdAt": Any<Date>,
      "id": 100,
      "loggedInAt": null,
      "roleId": 10,
      "updatedAt": null,
      "username": "Relation User",
    },
  ],
  "user_create": 0,
  "user_delete": 0,
  "user_read": 0,
  "user_update": 0,
}
`;
